plugins/fortinet_fortigate/icon_fortinet_fortigate/util/api.py

```diff
@@ -0,0 +1,104 @@
+        self.helper = Helpers(self.logger)
+
+    def call_api(
+        self, path: str, method: str = "GET", params: dict = None, json_data: dict = None
+    ) -> requests.Response:
+        try:
+            response = self.session.request(
+                method=method.upper(),
+                url=f"https://{self.host}/api/v2/cmdb/{path}",
+                verify=self.ssl_verify,
+                json=json_data,
+                params=params,
+            )
+            self.helper.http_errors(response.text, response.status_code)
+            if response.status_code > 500:
+                raise PluginException(preset=PluginException.Preset.SERVER_ERROR, data=response.text)
+
+            if 200 <= response.status_code < 300:
+                return response.json()
+
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=response.text)
+        except json.decoder.JSONDecodeError as e:
+            raise PluginException(preset=PluginException.Preset.INVALID_JSON, data=e)
+        except requests.exceptions.HTTPError as e:
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=e)
+
+    def create_address_object(self, name: str, host: str, address_type: str):
+        endpoint = "firewall/address"
+        payload = {"name": name, "type": address_type}
+        if address_type == "ipmask":
+            payload["subnet"] = host
+        elif address_type == "fqdn":
+            payload["fqdn"] = host
+        else:
+            payload["ip6"] = host
+            endpoint = "firewall/address6"
+        return self.call_api(endpoint, "POST", json_data=payload)
+
+    def delete_address_object(self, endpoint: str):
+        return self.call_api(endpoint, "DELETE")
+
+    def get_address_group(self, address_group_name: str, is_ipv6: bool):
+        endpoint = "firewall/addrgrp"
+
+        if is_ipv6:
+            endpoint = "firewall/addrgrp6"
+
+        result = self.call_api(
+            path=endpoint,
+            params={
+                "filter": f"name=@{address_group_name}",
+            },
+        )
+
+        groups = result.get("results")
+        if not len(groups) > 0:
+            raise PluginException(
+                cause=f"Could not find address group '{address_group_name}' in results.\n",
+                assistance=f"Please make sure the group '{address_group_name}' exists.\n",
+                data=result,
+            )
+
+        return groups[0]
+
+    def get_address_object(self, address_name: str):
+        try:
+            response_ipv4 = self.call_api(path=f"firewall/address/{address_name}")
+            if response_ipv4.get("http_status") == 200:
+                return response_ipv4
+        except PluginException:
+            self.logger.info(f"The specified object {address_name} was not found in domain and IPv4 address objects.")
+
+        response_ipv6 = self.call_api(path=f"firewall/address6/{address_name}")
```

Comment:
Can you please add Not found exception for IpV6 same as IPV4?

Reply:
@blaxminarayan-r7 Two separate endpoints are used to obtain address objects for IPv4 or IPv6. This method aims to get information about a given address object based on its name, which does not allow to clearly determine whether the searched object is for an IPv4 or IPv6 address. Therefore, we first look for the given object using the endpoint for IPv4 address objects, as these addresses are used more often. If it is not found and `call_api` method raises an exception, we catch it, log this exception and message without raising new exception, and then try to get the object using the endpoint for IPv6 address objects. If the object is still not found, `call_api` method will raise an exception. When we receive a response, but the `http_status` value is not 200, we raises the exception that the given object does not exist. Therefore, adding new, another Exception is not necessary.